bsp_diff/common/kernel/linux-intel-lts2021/0024-add-telemetry-for-pci-device-state.patch

```diff
@@ -0,0 +1,56 @@
+From a032f7751417c8fbabaa55f435d56c3ac8d3295c Mon Sep 17 00:00:00 2001
+From: Qi Zhang <qi1.zhang@intel.com>
+Date: Fri, 1 Sep 2023 21:48:50 +0800
+Subject: [PATCH] add telemetry for pci device state
+
+Tracked-On: OAM-112036
+Signed-off-by: Qi Zhang <qi1.zhang@intel.com>
+---
+ drivers/platform/x86/intel/pmc/core.c | 24 ++++++++++++++++++++++++
+ 1 file changed, 24 insertions(+)
+
+diff --git a/drivers/platform/x86/intel/pmc/core.c b/drivers/platform/x86/intel/pmc/core.c
+index 84a9a42220f2..bd8da742ea76 100644
+--- a/drivers/platform/x86/intel/pmc/core.c
++++ b/drivers/platform/x86/intel/pmc/core.c
+@@ -1836,6 +1836,27 @@ static int pmc_core_pkgc_show(struct seq_file *s, void *unused)
+ }
+ DEFINE_SHOW_ATTRIBUTE(pmc_core_pkgc);
+ 
++static int pmc_core_pci_device_state_show(struct seq_file *s, void *unused)
++{
++	struct pci_dev *dev = NULL;
++	u32 d3_state;
++	u16 pmcsr;
++
++	for_each_pci_dev(dev) {
++		pci_read_config_word(dev, dev->pm_cap + PCI_PM_CTRL, &pmcsr);
++		d3_state = ((pmcsr & PCI_PM_CTRL_STATE_MASK) ==
++					(__force int)PCI_D3hot) ? 1 : 0;
++
++		seq_printf(s, "pci %04x %04x %s %20.20s: ",
++					dev->vendor, dev->device, dev_name(&dev->dev),
++					dev_driver_string(&dev->dev));
++		seq_printf(s, " d3:%x\n", d3_state);
++	}
++
++	return 0;
++}
++DEFINE_SHOW_ATTRIBUTE(pmc_core_pci_device_state);
++
+ static bool pmc_core_pri_verify(u32 lpm_pri, u8 *mode_order)
+ {
+ 	int i, j;
+@@ -1980,6 +2001,9 @@ static void pmc_core_dbgfs_register(struct pmc_dev *pmcdev)
+ 				    pmcdev->dbgfs_dir, pmcdev,
+ 				    &pmc_core_substate_req_regs_fops);
+ 	}
++
++	debugfs_create_file("pci_device_state", 0644, dir, pmcdev,
```

Comment:
Since we implement SHOW attribute is only implemented, shall we change permission to 0444 from 0644

Reply:
done